sympy/geometry/point.py

```diff
@@ -477,29 +580,27 @@ def intersection(self, other):
         [Point2D(0, 0)]
 
         """
-        other = Point.pointify(other, ensure_point=False)
+        if not isinstance(other, (Point, GeometryEntity)):
```

Comment:
`Point` inherits from `GeometryEntity` so `isinstance(other, GeometryEntity)` will do. This shows up several times.